app/authorization.go

```diff
@@ -138,23 +138,18 @@ func (a *App) HasPermissionToTeam(askingUserId string, teamId string, permission
 	if teamId == "" || askingUserId == "" {
 		return false
 	}
+	teamMember, _ := a.GetTeamMember(teamId, askingUserId)
+	if teamMember != nil {
```

Comment:
For clarity, could we instead test `teamMember != nil && teamMember.DeleteAt == 0` for this block, and then just fall back to the existing `a.HasPermissionTo(askingUserId, permission)` at the bottom of the method instead of duplicating that check a second time?